test/testdata/infer/generics/isa_with_type_member.rb

```diff
@@ -0,0 +1,99 @@
+# typed: true
+
+class A
+  extend T::Generic
+  X = type_member
+end
+
+class B < A
+  extend T::Generic
+  X = type_member(fixed: String)
+end
+
+class Test
+  extend T::Sig
+
+  sig {params(x: A[T.untyped]).void}
+  def test1(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `B`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[T.untyped]`
+    end
+  end
+
+  sig {params(x: A[String]).void}
+  def test2(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `B`
+    when A
+      T.reveal_type(x) # error: Revealed type: `A[String]`
+    end
+  end
+
+  sig {params(x: T.nilable(A[Integer])).void}
+  def test3(x)
+    case x
+    when B
+      T.reveal_type(x) # error: Revealed type: `T.all(B, A[Integer])`
```

Comment:
Is this type conceptually bottom? If not, could we test what it means to have this type on hand? (i.e., what methods should we be allowed to call on it?)

Reply:
I had a long discussion with Dmitry in which he convinced me that it's good to not consider this type bottom. Methods that involve the conflicting type parameter will not be callable, but anything else will be fine.